### PR TITLE
Bump aeson

### PR DESCRIPTION
**Pull request by soiamsoNG**
Bump aeson to work with aeson 2.0 (default ordered KeyMap)
It also work with GHC 9.2.1 with configuration  allow-newer: all:all

### CODE REVIEW

File: heist.cabal [hunk 1]
```diff
@@ -162,7 +162,7 @@ Library
     Heist.Interpreted.Internal
 
   build-depends:
-    aeson                      >= 0.6     && < 1.6,
+    aeson                      >= 0.6     && < 3,
```
Comment (andreasabel) [inline heist.cabal:165]: `<3` isn't a good upper bound.  Please change to `<2.1`.

File: src/Heist/Splices/Json.hs [hunk 2]
```diff
@@ -89,7 +96,20 @@ boolToText b = if b then "true" else "false"
...
 
+#if MIN_VERSION_aeson(2,0,0)
+------------------------------------------------------------------------------
+findExpr :: Text -> Value -> Maybe Value
+findExpr t = go (T.split (=='.') t)
+  where
+    go [] !value     = Just value
+    go (x:xs) !value = findIn value >>= go xs
+      where
+        findIn (Object obj) = KM.lookup (K.fromText x) obj
```
Comment (andreasabel) [inline src/Heist/Splices/Json.hs:107]: This is the only line that is different and should be in the `#if`.  It seems superfluous to cut&paste the whole function.

File: src/Heist/Splices/Json.hs [hunk 3]
```diff
@@ -214,6 +234,22 @@ explodeTag = ask >>= go
         "value"    ## varAttrTag v valueTag
 
 
+#if MIN_VERSION_aeson(2,0,0)
+    --------------------------------------------------------------------------
+    goObject :: KM.KeyMap Value -> ReaderT Value (HeistT n n) Template
+    goObject obj = do
+        start <- genericBindings
+        let bindings = FI.ifoldl' bindKvp start  obj
```
Comment (andreasabel) [inline src/Heist/Splices/Json.hs:242]: There is ways to make the code `aeson-2` compatible with less duplication of code.

**Review by andreasabel — CHANGES_REQUESTED** [review]

**Review by andreasabel — APPROVED** [review]
Excellent, thanks a lot @soiamsoNG !

**Comment by andreasabel** [comment]
ping @mightybyte

**Comment by cydparser** [comment]
Thanks!

**Pull request merged**